lib/b3fs/src/hasher/collector.rs

```diff
@@ -40,16 +42,28 @@ impl HashTreeCollector for BufCollector {
 }
 
 impl BufCollector {
+    #[cfg(feature = "async")]
     pub async fn write_hash(
         &mut self,
         mut writer: impl tokio::io::AsyncWriteExt + Unpin,
-    ) -> Result<(), io::Error> {
+    ) -> Result<(), tokio::io::Error> {
         writer.write_all(&self.buffer).await?;
         writer.flush().await?;
         self.buffer.clear();
         Ok(())
     }
 
+    #[cfg(not(feature = "async"))]
```

Comment:
Is this needed? I think if you don't feature flag by default if not

Reply:
yeah, if the flag isn't there, it will conflict with the method declaration above.